File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0 || (x%10==0 && x!=0))
        {
            return false;
        }
        int revnumber=0;
        while(x>revnumber)
        {
            revnumber=revnumber*10+x%10;
            x/=10;
        }
        return x==revnumber || x==revnumber/10;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x%10==0 && x>0){return false;}
        int temp=0;
        while(temp<x)
        {
            temp=temp*10 + x%10;
            x/=10;
        }
        return temp==x || (temp/10)==x;
    }
}